Core/include/Acts/Seeding2/TripletSeedFinder.hpp

```diff
@@ -0,0 +1,229 @@
+    float minHelixDiameter2 = std::numeric_limits<float>::quiet_NaN();
+    float sigmapT2perRadius = std::numeric_limits<float>::quiet_NaN();
+    float multipleScattering2 = std::numeric_limits<float>::quiet_NaN();
+  };
+
+  explicit TripletSeedFinder(const DerivedConfig& config);
+
+  const DerivedConfig& config() const { return m_impl->config(); }
+
+  /// Create triplets from the bottom, middle, and top space points.
+  ///
+  /// @param spacePoints Space point container
+  /// @param spM Space point candidate to be used as middle SP in a seed
+  /// @param bottomDoublet Bottom doublet to be used for triplet creation
+  /// @param topDoublets Top doublets to be used for triplet creation
+  /// @param tripletTopCandidates Cache for triplet top candidates
+  void createTripletTopCandidates(
+      const SpacePointContainer2& spacePoints, const ConstSpacePointProxy2& spM,
+      const DoubletsForMiddleSp::Proxy& bottomDoublet,
+      DoubletsForMiddleSp::Range& topDoublets,
+      TripletTopCandidates& tripletTopCandidates) const {
+    m_impl->createTripletTopCandidates(spacePoints, spM, bottomDoublet,
+                                       topDoublets, tripletTopCandidates);
+  }
+
+  /// Create triplets from the bottom, middle, and top space points.
+  ///
+  /// @param spacePoints Space point container
+  /// @param spM Space point candidate to be used as middle SP in a seed
+  /// @param bottomDoublet Bottom doublet to be used for triplet creation
+  /// @param topDoublets Top doublets to be used for triplet creation
+  /// @param tripletTopCandidates Cache for triplet top candidates
+  void createTripletTopCandidates(
+      const SpacePointContainer2& spacePoints, const ConstSpacePointProxy2& spM,
+      const DoubletsForMiddleSp::Proxy& bottomDoublet,
+      DoubletsForMiddleSp::Subset& topDoublets,
+      TripletTopCandidates& tripletTopCandidates) const {
+    m_impl->createTripletTopCandidates(spacePoints, spM, bottomDoublet,
+                                       topDoublets, tripletTopCandidates);
+  }
+
+  /// Create triplets from the bottom, middle, and top space points.
+  ///
+  /// @param spacePoints Space point container
+  /// @param spM Space point candidate to be used as middle SP in a seed
+  /// @param bottomDoublet Bottom doublet to be used for triplet creation
+  /// @param topDoublets Top doublets to be used for triplet creation
+  /// @param tripletTopCandidates Cache for triplet top candidates
+  void createTripletTopCandidates(
+      const SpacePointContainer2& spacePoints, const ConstSpacePointProxy2& spM,
+      const DoubletsForMiddleSp::Proxy& bottomDoublet,
+      DoubletsForMiddleSp::Subset2& topDoublets,
+      TripletTopCandidates& tripletTopCandidates) const {
+    m_impl->createTripletTopCandidates(spacePoints, spM, bottomDoublet,
+                                       topDoublets, tripletTopCandidates);
+  }
+
+ private:
+  class ImplBase {
```

Comment:
Not important, but you could forward declare this if you're ok with moving the function calls into this class into the `.cpp` file (they're currently defined here in the header).